nx/vitest@^20.7: Cannot find module '@angular/build/private' when using yarn

### Current Behavior

When running the command below:

```bash
yarn nx run-many -t test --skip-nx-cache
```

For the project `nx-vitest-cant-find-module-ui`, the following error is produced:

```bash
yarn run v1.22.22
$ /.../nx-vitest-cant-find-module/node_modules/.bin/nx run-many -t test --skip-nx-cache

 NX   Running target test for project nx-vitest-cant-find-module-ui

   ✖  nx run nx-vitest-cant-find-module-ui:test
      failed to load config from /.../nx-vitest-cant-find-module/apps/nx-vitest-cant-find-module-ui/vite.config.mts
      
       NX   Cannot find module '@angular/build/private'
      
      Require stack:
      - /.../nx-vitest-cant-find-module/node_modules/@analogjs/vite-plugin-angular/src/lib/utils/devkit.js
      Pass --verbose to see the stacktrace.
```
**It only happens when I use Yarn. For example, the same commands work fine with pnpm. However, I need to use Yarn for this project.** 

### Expected Behavior

I expect the test command to run without error. Specifically, the configuration file at `vite.config.mts` should load correctly, and the module `@angular/build/private` should be resolved so that tests run smoothly.


### GitHub Repo

_No response_

### Steps to Reproduce

1. **Create a new workspace:**  
   Run the following command to create an Angular monorepo using Nx:
   ```bash
   npx --yes create-nx-workspace@20.7.2 nx-vitest-cant-find-module --preset angular-monorepo --appName nx-vitest-cant-find-module-ui --bundler esbuild --style scss --ssr false --e2eTestRunner playwright --ci skip --pm yarn
   ```
2. **Run the test command:**  
   Execute:
   ```bash
   yarn nx run-many -t test --skip-nx-cache
   ```
3. **Observe the error:**  
   The error about missing module `@angular/build/private` appears when running tests for the `nx-vitest-cant-find-module-ui` project.


### Nx Report

```shell
yarn run v1.22.22
$ /.../nx-vitest-cant-find-module/node_modules/.bin/nx report

 NX   Report complete - copy this into the issue template

Node           : 22.5.1
OS             : darwin-arm64
Native Target  : aarch64-macos
yarn           : 1.22.22

nx                     : 20.7.2
@nx/js                 : 20.7.2
@nx/eslint             : 20.7.2
@nx/workspace          : 20.7.2
@nx/angular            : 20.7.2
@nx/devkit             : 20.7.2
@nx/eslint-plugin      : 20.7.2
@nx/module-federation  : 20.7.2
@nx/playwright         : 20.7.2
@nx/rspack             : 20.7.2
@nx/vite               : 20.7.2
@nx/web                : 20.7.2
@nx/webpack            : 20.7.2
typescript             : 5.7.3
---------------------------------------
Registered Plugins:
@nx/playwright/plugin
@nx/eslint/plugin
---------------------------------------
Community plugins:
@analogjs/vite-plugin-angular : 1.14.1
@analogjs/vitest-angular      : 1.14.1
angular-eslint                : 19.3.0
---------------------------------------
Cache Usage: 0.00 B / 46.04 GB

✨  Done in 1.52s.
```

### Failure Logs

```shell
All relevant logs (as shown in the **Current Behavior** section) have been included above.
```

### Package Manager Version

Yarn: **v1.22.22**

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

This issue appears related to the `@analogjs/vite-plugin-angular` plugin attempting to load a private module from Angular’s build package, which may no longer be available or has been moved. Any insight or guidance on how to resolve this dependency error would be appreciated!

--- a/packages/angular/src/generators/utils/add-vitest.ts
+++ b/packages/angular/src/generators/utils/add-vitest.ts
@@ -1,5 +1,10 @@
-import { ensurePackage, type Tree } from '@nx/devkit';
+import {
+  addDependenciesToPackageJson,
+  ensurePackage,
+  type Tree,
+} from '@nx/devkit';
 import { nxVersion } from '../../utils/versions';
+import { getInstalledAngularDevkitVersion, versions } from './version-utils';
 
 export type AddVitestOptions = {
   name: string;
@@ -25,4 +30,18 @@ export async function addVitest(
     coverageProvider: 'v8',
     addPlugin: options.addPlugin ?? false,
   });
+
+  if (!options.skipPackageJson) {
+    const angularDevkitVersion =
+      getInstalledAngularDevkitVersion(tree) ??
+      versions(tree).angularDevkitVersion;
+
+    addDependenciesToPackageJson(
+      tree,
+      {},
+      { '@angular/build': angularDevkitVersion },
+      undefined,
+      true
+    );
+  }
 }

--- a/packages/angular/src/generators/utils/version-utils.ts
+++ b/packages/angular/src/generators/utils/version-utils.ts
@@ -8,6 +8,13 @@ import {
 import * as latestVersions from '../../utils/versions';
 import { angularVersion } from '../../utils/versions';
 
+export function getInstalledAngularDevkitVersion(tree: Tree): string | null {
+  return (
+    getInstalledPackageVersion(tree, '@angular-devkit/build-angular') ??
+    getInstalledPackageVersion(tree, '@angular/build')
+  );
+}
+
 export function getInstalledAngularVersion(tree: Tree): string {
   const pkgJson = readJson(tree, 'package.json');
   const installedAngularVersion =

